bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/AbstractExpression.java

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.StringTokenizer;
+import java.util.TimeZone;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <code>AbstractExpression</code> is an abstract implementation of {@link Expression} that provides common
+ * functionality to other concrete implementations of <code>Expression</code>
+ *
+ * @author Karel Goderis - Initial Contribution
+ *
+ */
+public abstract class AbstractExpression<E extends AbstractExpressionPart> implements Expression {
+
+    final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+
```

Comment:
most of the below fields (if not all - see somments below...) can and should be private. I'd use protected getters instead - they make it easier to make it thread-safe